File: codeforces/122A/py/122a.py
def is_lucky(num):
    while (num):
        mod = num % 10
        num = int(num / 10)
        if mod != 4 and mod != 7:
            return False
    return True

def almost_lucky(num):
    for i in range(4, num+1):
        if is_lucky(i) == True:
            if num % i == 0:
                return True
    return False
    

num = int(input())
if almost_lucky(num) == True:
    print("YES")
else:
    print("NO")

File: codeforces/122A/cpp/122a.cpp
#include <iostream>
using namespace std;

bool is_lucky(int num)
{
    while (num) {
        int mod = num % 10;
        if (mod != 4 && mod != 7)
            return false;
        num /= 10;
    }
    return true;
}

int main()
{
    int num;
    cin >> num;

    bool almost_lucky = false;
    for (int i = 4; i <= num; i++) {
        if (is_lucky(i)) {
            if (num % i == 0) {
                almost_lucky = true;
            }
        }
    }

    if (almost_lucky)
        cout << "YES" << endl;
    else
        cout << "NO" << endl;

    return 0;
}
